packages/preview-middleware-client/src/adp/dialogs/fragment.ts

```diff
@@ -0,0 +1,47 @@
+/** sap.ui.core */
+import XMLView from 'sap/ui/core/mvc/XMLView';
+import type UI5Element from 'sap/ui/core/Element';
+import Controller from 'sap/ui/core/mvc/Controller';
+
+/** sap.ui.rta */
+import type RuntimeAuthoring from 'sap/ui/rta/RuntimeAuthoring';
+
+import type AddFragment from '../controllers/AddFragment.controller';
+
+/**
+ * Initilizes "Add XML Fragment" functionality and adds a new item to the context menu
+ *
+ * @param rta Runtime Authoring
+ */
+export const initFragment = (rta: RuntimeAuthoring): void => {
+    const contextMenu = rta.getDefaultPlugins().contextMenu;
+
+    contextMenu.addMenuItem({
+        id: 'ADD_FRAGMENT',
+        text: 'Add: Fragment',
+        handler: async (overlays: UI5Element[]) => await handler(overlays, rta),
+        icon: 'sap-icon://attachment-html'
+    });
+};
+
+/**
+ * Handler for new context menu entry
+ *
+ * @param overlays Control overlays
+ * @param rta Runtime Authoring
+ */
+export async function handler(overlays: UI5Element[], rta: RuntimeAuthoring): Promise<void> {
```

Comment:
I think I need help understanding this one.
You are creating an empty view referencing an instance of your controller. In your controller init you then add a fragment with your dialog.
Why don't you just create a view already containing the content of your fragment?

Reply:
Changed in https://github.com/SAP/open-ux-tools/pull/1183/commits/e0a55f9df1453f0624d0b2c1283d97d6c880374d
Now the view is not empty from the start and we do not call this.loadFragment in the onInit method.

Reply:
The question stays, why not just having a view with all the content and completely getting rid of the fragment.

Your view at `src/adp/ui/AddFragment.view.xml` would be 
```xml
<mvc:View xmlns:mvc="sap.ui.core.mvc"
    xmlns="sap.m"
    xmlns:core="sap.ui.core"
    xmlns:f="sap.ui.layout.form">
    <Dialog id="addNewFragmentDialog" title="Add XML Fragment" contentWidth="500px" class="sapUiRTABorder">
        <!-- ... -->
    </Dialog>
</mvc:View>
```

And then you can simplify your handler function to 
```ts
export async function handler(overlays: UI5Element[], rta: RuntimeAuthoring): Promise<void> {
    const controller = new AddFragment('open.ux.preview.client.adp.controllers.AddFragment');
    controller.overlays = overlays;
    controller.rta = rta;

    await XMLView.create({
        viewName: 'open.ux.preview.client.adp.ui.AddFragment',
        controller
    });
}
```

PS: you might also want to add the variable assignments of `overlays` and `rta` to the constructor.

Reply:
Done in: https://github.com/SAP/open-ux-tools/pull/1183/commits/60916e4ac16947c1e325a6c4cf4474d79a5b7aec